### PR TITLE
[Refactor RetinaNet] Step 10, Unify call and loss reduction for Retin…

### PR DESCRIPTION
**Pull request by tanzhenyu**
…aNet.

This makes model.call return decoded boxes, use SUM reduction across OD models.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: keras_cv/models/object_detection/retina_net/retina_net.py [hunk 1]
```diff
@@ -167,15 +167,6 @@ def __init__(
             output_filters=9 * 4, bias_initializer="zeros"
         )
 
-        self.loss_metric = tf.keras.metrics.Mean(name="loss")
-        self.classification_loss_metric = tf.keras.metrics.Mean(
-            name="classification_loss"
-        )
-        self.box_loss_metric = tf.keras.metrics.Mean(name="box_loss")
-        self.regularization_loss_metric = tf.keras.metrics.Mean(
```
Comment (ianstenbit) [inline keras_cv/models/object_detection/retina_net/retina_net.py:175]: It looks like the regularization loss and the overall sum loss are not reintroduced anywhere -- am I missing something or does this PR remove those metrics?

File: keras_cv/models/object_detection/retina_net/retina_net.py [hunk 2]
```diff
@@ -245,20 +217,24 @@ def call(self, images, training=False):
 
         return box_pred, cls_pred
 
+    def call(self, images, training=None):
+        box_pred, cls_pred = self._forward(images, training=training)
+        if not training:
```
Comment (ianstenbit) [inline keras_cv/models/object_detection/retina_net/retina_net.py:222]: I'm not certain how this will work with functional subclassing -- it seems like we'll want to get rid of `_forward` in the functional subclass model (that's how I've done it in my implementation of RetinaNet as a functional subclass -- PR pending #1203 being merged)

I suppose we can override call in the functional subclass model to call super.call and then decode the deltas to real boxes?

**Review by ianstenbit — CHANGES_REQUESTED** [review]

File: keras_cv/models/object_detection/retina_net/retina_net.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/models/object_detection/retina_net/retina_net.py:175]: It's re-introduced in `compile(...weight_decay)`, just like RCNN models

File: keras_cv/models/object_detection/retina_net/retina_net.py (hunk 2 above)
Comment (tanzhenyu) [reply keras_cv/models/object_detection/retina_net/retina_net.py:222]: Discussed offline, let's try this in your PR

**Review by ianstenbit — APPROVED** [review]
